### PR TITLE
streamline HVAC_MODE string handling

### PR DESCRIPTION
**Pull request by aquini**
Remove unnecessary duplicated strings, by keeping them mode strings in an array.
No functional changes.

### CODE REVIEW

**Comment by aquini** [comment]
Steve, 

This is untested, so take it with a grain of salt. I'm still getting myself acquainted with the codebase.

-- Rafael

**Comment by smeisner** [comment]
The new strings must match the old strings exactly. This is a requirement of Home Assistant home automation system ... which this is going to be targeted to.

**Comment by aquini** [comment]
> The new strings must match the old strings exactly. This is a requirement of Home Assistant home automation system ... which this is going to be targeted to.

Which of the routines had them right:  hvacModeToMqttCurrMode() or strToHvacMode()?

File: app/src/tft.cpp [hunk 1]
```diff
@@ -220,47 +220,45 @@ void tftUpdateDisplay()
   }
 }
 
+const char *hvac_mode_str[NR_MODES] = {
+  "off", "auto", "heating", "cooling", "fan-only", "aux-heat", "error", "idle",
```
Comment (smeisner) [inline app/src/tft.cpp:224]: After studying the code more in-depth, I now recall why things seem to be duplicated. Home Assistant/MQTT requires the lower case strings and ending with "ing" listed in hvacModeToMqttCurrMode(). The string returned from hvacModeToString() is specifically for the user interface. They look nicer on the screen. How about adding another parameter to indicate what string the caller is looking for?

Comment (aquini) [reply app/src/tft.cpp:224]: Steve, 

2 quick clarification question:

a) is this the MQTT/Home assistant info I should follow for these string modes?; and
b) do you want the mode words with their first character capitalized for the UI?

Cheers!

Comment (smeisner) [reply app/src/tft.cpp]: > a) is this the MQTT/Home assistant info [1] I should follow for these string modes?
Yes!!
> b) do you want the mode words with their first character capitalized for the UI?
Yes!!
> 
> [1] https://www.home-assistant.io/integrations/climate.mqtt/

**Review by smeisner — APPROVED** [review]
Changes look good to me! I did a build and test and both the local UI and Home Assistant worked as expected!

**Pull request merged**